cookbook/asset/custom_version_strategy.rst

```diff
@@ -0,0 +1,205 @@
+.. index::
+   single: Asset; Custom Version Strategy
+
+How to use a Custom Version Strategy for Assets
+===============================================
+
+Out of the box, Symfony exposes configuration options which allow you to
+specify the assets version and the format of the versioned path. If you require
+more control, you need to create a custom version strategy.
+
+.. versionadded:: 2.7
+    The Asset component was introduced in Symfony 2.7.
+
+Default Package
+---------------
+
+The default package is the `''` package. It is used when you do not specify
+a package name in the :ref:`asset <reference-twig-function-asset>` Twig function.
+In order to override the version strategy used by the default package,
+it is necessary to add a compiler pass.
+
+This example shows how to integrate with `gulp-buster`_.
+
+Create Compiler Pass
+~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: php
+
+    // src/AppBundle/DependencyInjection/Compiler/OverrideAssetsDefaultPackagePass.php
+    namespace AppBundle\DependencyInjection\Compiler;
+
+    use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+    use Symfony\Component\DependencyInjection\ContainerBuilder;
+    use Symfony\Component\DependencyInjection\Reference;
+
+    class OverrideAssetsDefaultPackagePass implements CompilerPassInterface
+    {
+        public function process(ContainerBuilder $container)
+        {
+            $definition = $container->getDefinition('assets._default_package');
+            $definition->replaceArgument(1, new Reference('app.assets.buster_version_strategy'));
+        }
+    }
+
+The code above fetches the service definition of the default package, and replaces
+its 2nd argument (the version strategy).
+
+Register Compiler Pass
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: php
+
+    // src/AppBundle/AppBundle.php
+    namespace AppBundle;
+
+    use AppBundle\DependencyInjection\Compiler\OverrideAssetsDefaultPackagePass;
+    use Symfony\Component\DependencyInjection\ContainerBuilder;
+    use Symfony\Component\HttpKernel\Bundle\Bundle;
+
+    class AppBundle extends Bundle
+    {
+        public function build(ContainerBuilder $container)
+        {
+            parent::build($container);
+
+            // only register in prod environment
+            if ('prod' === $container->getParameter('kernel.environment')) {
+                $container->addCompilerPass(new OverrideAssetsDefaultPackagePass());
+            }
+        }
+    }
+
+See :doc:`/cookbook/service_container/compiler_passes` for more information
+on how to use compiler passes.
+
+Register Services
+~~~~~~~~~~~~~~~~~
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/services.yml
+        services:
+            app.assets.buster_version_strategy:
+                class: AppBundle\Asset\VersionStrategy\BusterVersionStrategy
+                arguments:
+                    - manifest_path: "%kernel.root_dir%/../busters.json"
```

Comment:
At the beginning of the article you say `This example shows how to integrate with `gulp-buster`_.` Mybe that's a good point to add more information about this specific "asset buster". You don't need to explain it in detail, but please add anything you consider useful. For example: "Gulp Buster requires to create one file caled bbla bla and have write permissions in the bla bla directory. Besides, the manifest.json directory must be formated like this and that. Here it is a small example of its contents ..."